src/process/ProcessManagerImpl.cpp

```diff
@@ -831,7 +884,6 @@ ProcessManagerImpl::maybeRunPendingProcesses()
 
             i->mImpl->run();
             mProcesses[i->mImpl->getProcessId()] = i;
-            ++gNumProcessesActive;
```

Comment:
Dumb question: how likely process IDs get recycled? Is it possible to have a race where we blindly clobber a process that exited before with `mProcesses[i->mImpl->getProcessId()] = i` (before calling the callback)?

Reply:
Unlikely but fair. I'll add a guard and consider it a failed startup on the new process (which there's plenty of upper machinery to retry in any cases that matter).